### PR TITLE
Adding meta-documentation and updating github.io ARXIVNG-1624

### PR DESCRIPTION
**Pull request by JaimieMurdock**
In this PR I add the meta-documentation, correct an issue in arxiv.base.agent and update https://arxiv.github.io/arxiv-base

The intersphinx modules are not working, I suspect this is due to missing `objects.inv` files. I'll open another ticket to review those:

```
loading intersphinx inventory from https://cul-it.github.io/arxiv-base/objects.inv...
WARNING: failed to reach any of the inventories with the following issues:
WARNING: intersphinx inventory 'https://cul-it.github.io/arxiv-base/objects.inv' not fetchable due to <class 'requests.exceptions.HTTPError'>: ('intersphinx inventory %r not fetchable due to %s: %s', 'https://cul-it.github.io/arxiv-base/objects.inv', <class 'requests.exceptions.HTTPError'>, HTTPError(...))
```

### CODE REVIEW

File: docs/source/arxiv/metadocs.rst [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+    
+    * :ref:`genindex`
+    * :ref:`modindex`
+    * :ref:`search`
+
+If you have an issue with ModuleNotFoundError, then the paths in the conf.py need to be changed. Add the following to the top of docs/conf.py::
+
+
+.. code-block:: python
+    :caption: conf.py
```
Comment (erickpeirson) [inline docs/source/arxiv/metadocs.rst]: Might want to double check that this indentation is ok. Sometimes Sphinx is picky about 3 vs 4 spaces

File: docs/source/arxiv/metadocs.rst [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+
+    import os
+    import sys
+    sys.path.insert(0, os.path.abspath('.'))
+    sys.path.append(os.path.abspath('..'))
+
+Configuring GitHub pages
+''''''''''''''''''''''''''
```
Comment (erickpeirson) [inline docs/source/arxiv/metadocs.rst]: Sphinx may warn about this

File: docs/source/arxiv/metadocs.rst (hunk 1 above)
Comment (JaimieMurdock) [reply docs/source/arxiv/metadocs.rst]: sphinx is a 3 space weirdness for the `.. directive::` syntax, but then 4 for the block itself. Who knows :confused:

Comment (JaimieMurdock) [reply docs/source/arxiv/metadocs.rst]: The warning was about a preceding `::` which messed with indentation levels. Fixed.

**Pull request merged**